.palantir/revapi.yml

```diff
@@ -1,6 +1,12 @@
 versionOverrides:
   org.apache.iceberg:iceberg-api:apache-iceberg-0.14.0: "0.14.0"
 acceptedBreaks:
+  apache-iceberg-0.14.0:
+    org.apache.iceberg:iceberg-api:
+    - code: "java.class.defaultSerializationChanged"
```

Comment:
Just to confirm that this doesn't change the the partition spec string representation, right? That seems to be the case from my quick scan of the PR.

Want to see if it affects the java serialization of `BaseContentScanTask`. 
```
  BaseContentScanTask(F file, String schemaString, String specString, ResidualEvaluator residuals)
```

This is related to a pending task in the Flink source checkpointing state (both current `FlinkSource` and the new FLIP-27 `IcebergSource`).
https://github.com/apache/iceberg/issues/1698

Reply:
Yeah, this doesn't affect it. The change was that now an array of `Function` is serialized rather than an array of `Transfrom`. That changes what gets serialized, but Java serialization still works just fine. All of the `Function` instances returned by `bind` should be `Serializable` (a good thing to check!).

Reply:
I did not initially notice actual functions returned by `bind` are serializable. Still seems like a fragile space if we don't enforce it on the API level.

Reply:
I added `SerializableFunction` and made `bind` return it.

Reply:
And I added serialization tests for all supported transform/type combinations.

Reply:
Resolving this thread then.